PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx

```diff
@@ -129,131 +143,201 @@ struct HfCandidateCreatorSigmac0plusplus {
+                      /* 2-prong specific columns */
+                      candLc.px(), candLc.py(), candLc.pz(),
+                      trackSoftPi.px(), trackSoftPi.py(), trackSoftPi.pz(),
+                      candLc.globalIndex(), trackSoftPi.globalIndex(),
+                      candLc.hfflag(),
+                      /* Σc0,++ specific columns */
+                      chargeSigmac,
+                      statusSpreadMinvPKPiFromPDG, statusSpreadMinvPiKPFromPDG);
+    } /// end loop over Λc+ → pK-π+ (and charge conj.) candidates
+  }   /// end makeSoftPiLcPair
+
+  /// @brief function to loop over candidate soft pions and, for each of them, over candidate Λc+ for Σc0,++ → Λc+(→pK-π+) π- candidate reconstruction
+  /// @param collision is a o2::aod::Collisions
+  /// @param trackSoftPi is the track (with dcaXY, dcaZ information)of a candidate soft-pion in the collision
+  /// @param tracks are the tracks (with dcaXY, dcaZ information) → soft-pion candidate tracks
+  /// @param candidates are 3-prong candidates satisfying the analysis selections for Λc+ → pK-π+ (and charge conj.)
+  template <bool withTimeAssoc, typename TRK>
+  void createSigmaC(aod::Collisions::iterator const& collision,
+                    TRK const& trackSoftPi,
+                    aod::TracksWDcaExtra const& tracks,
+                    CandidatesLc const& candidates,
+                    aod::BCsWithTimestamps const& bcWithTimeStamps)
+  {
+
+    auto thisCollId = collision.globalIndex();
+
+    histos.fill(HIST("hCounter"), 2);
+    /// keep only soft-pion candidate tracks
+    /// if not selected, skip it and go to the next one
+    if (!softPiCuts.IsSelected(trackSoftPi)) {
+      return;
+    }
+    /// dcaXY, dcaZ selections
+    /// To be done separately from the others, because for reassigned tracks the dca must be recalculated
+    /// TODO: to be properly adapted in case of PV refit usage
+    if (trackSoftPi.collisionId() == thisCollId) {
+      /// this is a track originally assigned to the current collision
+      /// therefore, the dcaXY, dcaZ are those already calculated in the track-propagation workflow
+      if (std::abs(trackSoftPi.dcaXY()) > softPiDcaXYMax || std::abs(trackSoftPi.dcaZ()) > softPiDcaZMax) {
+        return;
+      }
+    } else {
+      /// this is a reassigned track
+      /// therefore we need to calculate the dcaXY, dcaZ with respect to this new primary vertex
+      auto bc = collision.bc_as<o2::aod::BCsWithTimestamps>();
+      initCCDB(bc, runNumber, ccdb, isRun2Ccdb ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2Ccdb);
+      auto trackParSoftPi = getTrackPar(trackSoftPi);
+      o2::gpu::gpustd::array<float, 2> dcaInfo{-999., -999.};
+      o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackParSoftPi, 2.f, noMatCorr, &dcaInfo);
+      if (std::abs(dcaInfo[0]) > softPiDcaXYMax || std::abs(dcaInfo[1]) > softPiDcaZMax) {
+        return;
+      }
+    }
```

Comment:
In principle this is only needed in case you use the reassociated tracks, you could add a 
```
if constexpr (withTimeAssoc) {
} else {
}
```
To skip it in case you don't use reassociated tracks